sdl2promela/stop_condition/translator.py

```diff
@@ -919,7 +1103,19 @@ def _generate_filter_out_alternative(
 ) -> promela.Alternative:
     builder = promelaBuilder.AlternativeBuilder(promela.BlockType.BLOCK)
 
-    expressions = [_generate(context, s.expression) for s in statements]
+    expressions = []
+    for index in range(len(statements)):
+        context.precheck_expressions = []
+        expression = _generate(context, statements[index].expression)
```

Comment:
I think this needs a comment of intent for future generations

Reply:
function extracted, comments added